contracts/BaseStrategy.sol

```diff
@@ -608,26 +609,23 @@ abstract contract BaseStrategy {
         emit Harvested(profit, loss, debtPayment, debtOutstanding);
     }
 
-    /**
-     * Liquidate as many assets as possible to `want`, irregardless of
-     * slippage, up to `_amountNeeded`. Any excess should be re-invested
-     * here as well.
-     */
-    function liquidatePosition(uint256 _amountNeeded) internal virtual returns (uint256 _amountFreed);
-
     /**
      * @notice
      *  Withdraws `_amountNeeded` to `vault`.
      *
      *  This may only be called by the Vault.
      * @param _amountNeeded How much `want` to withdraw.
+     * @return loss Any realized losses
      */
-    function withdraw(uint256 _amountNeeded) external {
+    function withdraw(uint256 _amountNeeded) external returns (uint256 loss) {
         require(msg.sender == address(vault), "!vault");
         // Liquidate as much as possible to `want`, up to `_amount`
-        uint256 amountFreed = liquidatePosition(_amountNeeded);
+        uint256 amountFreed;
+        (amountFreed, loss) = liquidatePosition(_amountNeeded);
         // Send it directly back (NOTE: Using `msg.sender` saves some gas here)
         want.transfer(msg.sender, amountFreed);
```

Comment:
should we do instead
`want.transfer(msg.sender, Min(amountNeeded, amountFreed))` 

for the transfer, in case some strategy borks the accounting?

Reply:
No, I would say that's a bad strategy. See
https://github.com/iearn-finance/yearn-vaults/pull/140/files#diff-517cca2fbf046cbfefd26edb1fec816e5b2a5571fd99927dc81d6b4be60cd2a7R443

Reply:
I guess I could add an assertion for this though...

Reply:
yeah, i was thinking of how to mitigate if the invariant is not maintained, and i don't like adding a revert or require check also since it may lock up stuff, but i guess sending more than needed is still not the end of the world since withdraw on vault also checks for amount of shares, those "additional" funds just are transferred to the vault i guess.

Reply:
This is exactly right, looks like it is protected! Won't add the require statement.